src/import/README.md

```diff
@@ -0,0 +1,134 @@
+| rest-on-couch-home/eln/nmr/to_process | drop here the files to be processed                                                  |
+| rest-on-couch-home/eln/nmr/processed  | find here the processed files                                                        |
+| rest-on-couch-home/eln/nmr/errored    | find here the errored files                                                          |
+
+## import.js
+
+`import.js` will exports an async function that will be used to process files to import.
+`module.exports = async function nmrImport(ctx, result) {}`
+
+"Important note": filename HAS to be unique for a specific kind of data otherwise the existing file will be replaced !!
+This is true even if the field `reference` is different.
+
+The importation procedure has a special `ctx` in which the following variables are available:
+
+- filename: the name of the file to import
+- fileDir: the path to the file to import
+- fileExt: the extension of the file
+- couch: an instance of Couch
+
+It also has the following functions:
+
+- getContents(encoding = null, cache = true): allows to retrieve the content of the file. if the encoding is 'utf8' you get back a String that you can directly use, otherwise you get a Buffer.
+
+The result object may have the following properties:
+
+- kind: the kind of entry (eg 'sample')
+- id: the unique ID of the record in couchDB, the $ID field -(eg ['abc','def'])
+- owner: the owner of the entry if it does not exists, it must be an email (eg 'a@a.com')
+- reference = a unique ID that will decide if the attachment is added or replaced
+- field: name of the field in which the attachment will be saved (eg 'jcamp')
+- jpath: in which array should the attachment be saved (eg ['spectra', 'nmr'])
+- content_type: the mimetype of the attachment (eg 'chemical/x-jcamp-dx');
+- metadata: an object with all the properties containing metadata (eg {solvent:'CDCl3', frequency:400})
+- content: corresponds to the full entry. If you put an object in the `content` property it will be merged with the existing
+  data
+
+Result object also has the following functions:
+
+- addGroup(group): allows to add a group to the entry (eg 'addGroup(group1)');
+- addGroups(groups): allows to add many groups at once to the entry (eg addGroups(['group2', 'group3']))
+- skipAttachment(): useful if you want to add manually attachments. This could be the case if you need to change the
+  content of the file or if you want to attach more than one file at the same time.
+- skipMetadata()
+- addAttachment()
+
+## FAQ
+
+### Unique filename
+
+Script that generates unique filenames forcing all the files to be imported.
+
+```js
+module.exports = async function import(ctx, result) {
+    result.skipAttachment(); // we don't use the default importation procedure
+    var newFilename=ctx.filename.replace(/(._)\.(._)/,'$1.'+Date.now()+'.$2');
```

Comment:
If I remember correctly if you do not specify `skipMetadata` and no metadata is present it will throw